cli/build_run_result_registerer.go

```diff
@@ -0,0 +1,139 @@
+package cli
+
+import (
+	"errors"
+	"time"
+
+	"github.com/bitrise-io/bitrise/analytics"
+	"github.com/bitrise-io/bitrise/bitrise"
+	"github.com/bitrise-io/bitrise/exitcode"
+	"github.com/bitrise-io/bitrise/models"
+	"github.com/bitrise-io/bitrise/tools/timeoutcmd"
+	"github.com/bitrise-io/go-utils/colorstring"
+	"github.com/bitrise-io/go-utils/pointers"
+	coreanalytics "github.com/bitrise-io/go-utils/v2/analytics"
+	stepmanModels "github.com/bitrise-io/stepman/models"
+	log "github.com/sirupsen/logrus"
+)
+
+type BuildRunResultRegisterer struct {
```

Comment:
I think the correct word for this is `Registrar`, but I'd also be happy with something like `BuildRunResultCollector`